pkg/runtime/wfengine/activity.go

```diff
@@ -168,11 +173,16 @@ func (a *activityActor) executeActivity(ctx context.Context, actorID string, nam
 	//       introduce some kind of heartbeat protocol to help identify such cases.
 	callback := make(chan bool)
 	wi.Properties[CallbackChannelProperty] = callback
+	start := time.Now()
 	wfLogger.Debugf("Activity actor '%s': scheduling activity '%s' for workflow with instanceId '%s'", actorID, name, wi.InstanceID)
 	if err = a.scheduler(ctx, wi); err != nil {
 		if errors.Is(err, context.DeadlineExceeded) {
+			// Activity execution failed with recoverable error, record metrics.
+			diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, "dapr", diag.Activity, diag.StatusRetryable, 0)
```

Comment:
Define another constant in workflow_metrics.go `ComponentName: "dapr"` and use that everywhere instead of using "dapr" string at all places for componentKey